### PR TITLE
Update Kotlin to v1.8.0

### PR DESCRIPTION
**Pull request by 3flex**
https://kotlinlang.org/docs/whatsnew18.html
https://github.com/JetBrains/kotlin/releases/tag/v1.8.0

### CODE REVIEW

**Comment by 3flex** [comment]
:boom:

**Comment by 3flex** [comment]
Blocked on https://github.com/tschuchortdev/kotlin-compile-testing/issues/336

We might need to switch to running all tests via CLI and/or Gradle for the compiler plugin to avoid being blocked by this dependency when Kotlin is updated.

File: detekt-compiler-plugin/build.gradle.kts [hunk 1]
```diff
@@ -42,6 +42,11 @@ javaComponent.withVariantsFromConfiguration(configurations["shadowRuntimeElement
     skip()
 }
 
+tasks.test {
+    // https://github.com/detekt/detekt/issues/5646
+    enabled = false
```
Comment (3flex) [inline detekt-compiler-plugin/build.gradle.kts:47]: :eyes:

**Review by cortinico — APPROVED** [review]

**Comment by tokuhirom** [comment]
Is there a plan to release new version includes this commit? I want  to use detekt with kotlin-1.8.x

**Comment by 3flex** [comment]
You can either:
1. Use a snapshot version of detekt which has this commit included.
2. Update your project to Kotlin 1.8 but use the existing detekt version. As long as you're not using any new Kotlin 1.8 language features this should work.

**Comment by eboudrant** [comment]
> Update your project to Kotlin 1.8 but use the existing detekt version. As long as you're not using any new Kotlin 1.8 language features this should work.

What do you mean by that? We updated to Kotlin 1.8.10 (with Detekt 1.22.0) but we still see the error, we're not using any new Kotlin 1.8 language features.

**Comment by 3flex** [comment]
> we still see the error

What error do you see? Consider opening a new issue so we can assist.

**Comment by 3flex** [comment]
Hi, please open a new issue for this, thanks!

**Comment by pexa-agopalakrishnan** [comment]
> I have the same problem as described above, after trying to switch to kotlin 1.8 with same detekt version 1.22.0 I see:
> 
> ```
> * What went wrong:
> Execution failed for task ':archiv-impl:detekt'.
> > Analyzing /builds/path/to/sourceFolder/src/main/kotlin/de/company/package/info/CacheInfo.kt led to an exception.
>   Location: io.gitlab.arturbosch.detekt.rules.bugs.ImplicitDefaultLocale.checkStringFormatting(ImplicitDefaultLocale.kt:70)
>   The original exception message was: 'org.jetbrains.kotlin.psi.KtExpression org.jetbrains.kotlin.resolve.calls.util.CallUtilKt.getCalleeExpressionIfAny(org.jetbrains.kotlin.psi.KtElement)'
>   Running detekt '1.22.0' on Java '17.0.6+10' on OS 'Linux'
>   If the exception message does not help, please feel free to create an issue on our GitHub page.
> ```

yes, im also facing the same issue after upgrading

**Pull request merged**